### PR TITLE
Newsudicerts

### PR DESCRIPTION
**Pull request by joylatten**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -265,7 +258,20 @@ func generateManifestCreds(newUUID string) error {
...
-	err = os.WriteFile(filepath.Join(trustDir, "manifest.key"), keyPEM, 0600)
+	keyPEM, err := os.Create(filepath.Join(trustDir, "manifest.key"))
+	if err != nil {
+		return err
+	}
+	pkcs8, err := x509.MarshalPKCS8PrivateKey(privKey)
+	if err != nil {
+		return err
+	}
+	pem.Encode(keyPEM, &pem.Block{Type: "PRIVATE KEY", Bytes: pkcs8})
```
Comment (hallyn) [inline main.go:269]: pem.Encode returns an error

Comment (joylatten) [reply main.go]: Fixed in main.go and sudi.go

File: sudi.go [hunk 2]
```diff
@@ -156,7 +156,10 @@ func doSudiCert(VMname string) error {
 	if err != nil {
 		return err
 	}
-	pem.Encode(certPEM, &pem.Block {Type: "CERTIFICATE", Bytes: serverCert})
+	err = pem.Encode(certPEM, &pem.Block {Type: "CERTIFICATE", Bytes: serverCert})
```
Comment (hallyn) [inline sudi.go:159]: It would have been better to just do this in the first patch, but I guess it's not a big deal.

**Review by hallyn — APPROVED** [review]

File: sudi.go (hunk 2 above)
Comment (joylatten) [reply sudi.go:159]: true! I should have squashed them. Sorry.

**Pull request merged**